File: Module.Frontend.TPM/Client/model/tpm/brand/HistoricalBrand.js
﻿Ext.define('App.model.tpm.brand.HistoricalBrand', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'Brand',
    fields: [
        { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'Name', type: 'string', hidden: false, isDefault: true },
        { name: 'Brand_code', type: 'string', hidden: false, isDefault: true },
        { name: 'Segmen_code', type: 'string', hidden: false, isDefault: true }
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalBrands',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});

File: Module.Persist.TPM/Model/History/HistoricalBrand.cs
using Core.History;
using Module.Persist.TPM.Model.TPM;

namespace Module.Persist.TPM.Model.History
{
    [AssociatedWith(typeof(Brand))]
    public class HistoricalBrand : BaseHistoricalEntity<System.Guid>
    {
        public string Name { get; set; }
        public string Brand_code { get; set; }
        public string Segmen_code { get; set; }
    }
}
